control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -1518,3 +1522,16 @@ func (a *genericListAccessor) len() int {
 func (a *genericListAccessor) item(i int) client.Object {
 	return (a.items.Index(i).Addr().Interface()).(client.Object)
 }
+
+func (r *reconciler) reconcileStorage(ctx context.Context, hcp *hyperv1.HostedControlPlane, releaseImage *releaseinfo.ReleaseImage) []error {
+	var errs []error
+
+	cr := manifests.CSISnapshotController()
+	if _, err := r.CreateOrUpdate(ctx, r.client, cr, func() error {
+		storage.ReconcileCSISnapshotController(cr)
+		return nil
+	}); err != nil {
+		errs = append(errs, fmt.Errorf("failed to reconcile CSISnapshotController : %w", err))
```

Comment:
why is this appending here and returning later rather than just returning? it seems it's the only err.

Reply:
I plan to add other storage CRs in a subsequent PR, namely `Storage` for cluster-storage-operator.